src/sql/src/plan/query.rs

```diff
@@ -4964,7 +4964,9 @@ struct AggregateTableFuncVisitor<'a> {
     scx: &'a StatementContext<'a>,
     aggs: Vec<Function<Aug>>,
     within_aggregate: bool,
-    tables: HashMap<TableFunction<Aug>, String>,
+    // We maintain the `Vec` for deterministic ordering and the `HashMap` for fast lookups.
+    tables: Vec<(TableFunction<Aug>, String)>,
+    tables_map: HashMap<TableFunction<Aug>, String>,
```

Comment:
My immediate Q is whether a `BTreeMap` would work and if not what should we fix so that it does.